hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/HazelcastCacheRegionFactory.java

```diff
@@ -58,6 +56,14 @@ public EntityRegion buildEntityRegion(final String regionName, final Properties
                 new IMapRegionCache(regionName, instance, properties, metadata));
     }
 
+    public NaturalIdRegion buildNaturalIdRegion(final String regionName, final Properties properties,
+                                                final CacheDataDescription metadata)
+            throws CacheException {
+        /* Natural id regions are never versioned, so pass in null for metadata */
```

Comment:
I've opened [hazelcast/hazelcast-hibernate#4](https://github.com/hazelcast/hazelcast-hibernate/pull/4) with this change. However, I didn't do this part this way. There's no need to pass `null` like this. I watched with a debugger and the `CacheDataDescription` always has `isVersioned()` returning `false`, which results in the `RegionCache` correctly skipping the version checks.

Given that Hibernate is doing the right thing all on its own, it seems better to me to pass the `CacheDataDescription` through to the cache. Currently, you get exactly the behavior you commented on here. But should that ever change in Hibernate, the cache should still "just work".